packages/core-database/lib/wallet-manager.js

```diff
@@ -376,6 +376,41 @@ module.exports = class WalletManager {
     return data
   }
 
+  /**
+   * Determine the list of active delegates for the given round.
+   * @param  {Number} maxDelegates
+   * @return {Array}
+   */
+  determineActiveDelegates (maxDelegates) {
+    const mapWallets = wallets => wallets.map(wallet => ({
+      publicKey: wallet.vote || wallet.publicKey,
```

Comment:
What is the point of having a logic like "use the public key of the delegate the wallet has voted for (wallet.vote), or if that is not set, then use wallet's own public key"? Looks wrong to me.

Also, below we filter the array to leave only entries with `wallet.vote` evaluating to true, this means the expression is identical to just `publicKey: wallet.vote`.

Reply:
Wallets that get mapped if they are a filler don't have votes themselves so they only have their `publicKey` property.